### PR TITLE
Update MFEM and Axom

### PR DESCRIPTION
**Pull request by white238**
This includes a bunch of CUDA build fixes in Axom.

Updates to the MFEM 4.5.2 release candidate.

### CODE REVIEW

File: scripts/spack/packages/mfem/package.py [hunk 1]
```diff
@@ -7,19 +7,18 @@
 import shutil
 import sys
 
-from spack import *
+from spack.package import *
```
Comment (jamiebramwell) [inline scripts/spack/packages/mfem/package.py:10]: any word on MFEM adding a cmake-based variant?

**Review by jamiebramwell — APPROVED** [review]

File: scripts/spack/packages/mfem/package.py (hunk 1 above)
Comment (white238) [reply scripts/spack/packages/mfem/package.py:10]: none that i have heard.

File: scripts/spack/configs/config.yaml [hunk 2]
```diff
@@ -130,7 +126,7 @@ config:
   # If set to 4, for example, `spack install` will run `make -j4`.
   # If not set, all available cores are used by default.
   # for uberenv, limit build_jobs to 8
```
Comment (chapman39) [inline scripts/spack/configs/config.yaml:128]: comment says 8 is limit but switched to 16

**Review by chapman39 — APPROVED** [review]

File: scripts/spack/packages/axom/package.py [hunk 3]
```diff
@@ -7,13 +7,12 @@
 import socket
 from os.path import join as pjoin
 
-from spack import *
+from spack.package import *
```
Comment (chapman39) [inline scripts/spack/packages/axom/package.py:10]: adding `.package` is needed now?

Comment (white238) [reply scripts/spack/packages/axom/package.py:10]: I'm pretty sure that is the new spack standard. I just copy what is in the newer packages.

**Pull request merged**